cdap-app-fabric/src/main/java/co/cask/cdap/app/runtime/ProgramController.java

```diff
@@ -109,7 +109,9 @@
 
     State(ProgramRunStatus runStatus) {
       this.runStatus = runStatus;
-      this.programStatus = ProgramRunStatus.RUNNING == runStatus ? ProgramStatus.RUNNING : ProgramStatus.STOPPED;
+      this.programStatus = ProgramRunStatus.STARTING == runStatus || ProgramRunStatus.RUNNING == runStatus ?
```

Comment:
So the reason this had to be added was because all of the tests use `manager.waitForState(true)`, which looks to this to see if the program is running. If we don't do this, then `waitForState` methods may fail, since there is some time between `STARTING` and `ALIVE`.

So, either we:
1. increase the timeout of `waitForStatus`
2. More preferably, switch to checking the run records for a `ProgramRunStatus.RUNNING` when we check for waitForStatus.